compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -319,6 +319,14 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         targets: &SwitchTargets,
     ) {
         let discr = self.codegen_operand(bx, discr);
+        let discr_value = discr.immediate();
+        // If our discriminant is a constant we can branch directly
+        if let Some(const_discr) = bx.const_to_opt_u128(discr_value, true) {
```

Comment:
Why is `sign_ext` true?

Reply:
I wasn't quite sure what the right option there was. Should it be `false` or depend on the type of the discr?

Reply:
Yeah, I think it depends on the type of discr? I really don't know though -- I recommend looking at how these switch targets are evaluated.

Reply:
I'll come back to it once perf has gone through for my two MIR-opt based approaches. They should allow us to waste less time emitting dead bbs to LLVM.